### PR TITLE
feat(labels): add part-of selector label

### PR DESCRIPTION
**Pull request by andrewazores**
- **feat(labels): add part-of selector label**
- **chore(service): rename ports to remove 'cryostat-' prefix**

Related to https://github.com/cryostatio/cryostat-openshift-console-plugin/issues/2

### CODE REVIEW

**Comment by andrewazores** [comment]
The `ct.bash --upgrade` tests fail with:

```
>>> helm upgrade cryostat-8sjfi4ytca charts/cryostat --namespace helm-test --reuse-values --wait --timeout=600s
Error: UPGRADE FAILED: cannot patch "cryostat-8sjfi4ytca-v4" with kind Deployment: Deployment.apps "cryostat-8sjfi4ytca-v4" is invalid: spec.selector: Invalid value: v1.LabelSelector{MatchLabels:map[string]string{"app.kubernetes.io/component":"cryostat", "app.kubernetes.io/instance":"cryostat-8sjfi4ytca", "app.kubernetes.io/name":"cryostat", "app.kubernetes.io/part-of":"cryostat"}, MatchExpressions:[]v1.LabelSelectorRequirement(nil)}: field is immutable && cannot patch "cryostat-8sjfi4ytca-v4-db" with kind Deployment: Deployment.apps "cryostat-8sjfi4ytca-v4-db" is invalid: spec.selector: Invalid value: v1.LabelSelector{MatchLabels:map[string]string{"app.kubernetes.io/component":"db", "app.kubernetes.io/instance":"cryostat-8sjfi4ytca", "app.kubernetes.io/name":"cryostat", "app.kubernetes.io/part-of":"cryostat"}, MatchExpressions:[]v1.LabelSelectorRequirement(nil)}: field is immutable && cannot patch "cryostat-8sjfi4ytca-v4-storage" with kind Deployment: Deployment.apps "cryostat-8sjfi4ytca-v4-storage" is invalid: spec.selector: Invalid value: v1.LabelSelector{MatchLabels:map[string]string{"app.kubernetes.io/component":"storage", "app.kubernetes.io/instance":"cryostat-8sjfi4ytca", "app.kubernetes.io/name":"cryostat", "app.kubernetes.io/part-of":"cryostat"}, MatchExpressions:[]v1.LabelSelectorRequirement(nil)}: field is immutable
```

tl;dr the label selectors are immutable, so upgrading the Deployment in-place can't happen. I think this is kind of failure is OK so long as it only happens on semver major versions when breakages are expected (see https://github.com/cryostatio/cryostat-helm/pull/202).

```bash
$ bash ct.bash 2>&1 | tee ct.log
$ bash ct.bash --upgrade 2>&1 | tee ct-upgrade.log
$ diff ct.log ct-upgrade.log
```

**Review by ebaron — COMMENTED** [review]
We already have some unreleased label selector changes right? So I think the upgrade should still work after we renamed the deployment here like we did in the operator?

**Comment by ebaron** [comment]
> ^ #197 and #202
> 
> I think a 3.0 -> 4.0 upgrade will work because of this renaming, but if I understand how the upgrade test works correctly, it's trying to upgrade from main to this PR - so essentially 4.0 to 4.0, and there is no renaming happening between versions, hence the apparent immutability conflict.

Yup that sounds totally fine to me.

**Review by ebaron — APPROVED** [review]
